packages/tools/src/stateManagement/segmentation/polySeg/canComputeRequestedRepresentation.ts

```diff
@@ -0,0 +1,122 @@
+>([
+  [
+    SegmentationRepresentations.Labelmap,
+    new Set([
+      SegmentationRepresentations.Surface,
+      SegmentationRepresentations.Contour,
+    ]),
+  ],
+  [
+    SegmentationRepresentations.Contour,
+    new Set([
+      SegmentationRepresentations.Labelmap,
+      SegmentationRepresentations.Surface,
+    ]),
+  ],
+  [
+    SegmentationRepresentations.Surface,
+    new Set([SegmentationRepresentations.Labelmap]),
+  ],
+]);
+
+/**
+ * Determines whether the requested representation can be computed, based on
+ * the existing representation types and available conversion paths.
+ *
+ * @param segmentationRepresentationUID - The UID of the desired segmentation representation.
+ * @returns true if the requested representation can be computed, otherwise false.
+ */
+function canComputeRequestedRepresentation(
+  segmentationRepresentationUID: string
+): boolean {
+  const representationInfo = findSegmentationRepresentationByUID(
+    segmentationRepresentationUID
+  );
+
+  if (!representationInfo?.segmentationRepresentation) {
+    return false;
+  }
+
+  const { segmentationRepresentation } = representationInfo;
+  const { type: representationType, polySeg } = segmentationRepresentation;
+
+  if (!polySeg || !polySeg.enabled) {
+    return false;
+  }
+
+  const { representationData } = getSegmentation(
+    segmentationRepresentation.segmentationId
+  );
+
+  const existingRepresentationTypes =
+    getExistingRepresentationTypes(representationData);
+
+  return existingRepresentationTypes.some((existingRepresentationType) =>
+    canConvertFromTo(existingRepresentationType, representationType)
+  );
+}
+
+/**
+ * Retrieves the existing representation types for the given representationData
+ * by verifying the validity of each representation type.
+ *
+ * @param representationData - The representation data
+ * @returns supportedTypes - An array of valid representation types
+ */
+function getExistingRepresentationTypes(
+  representationData: SegmentationRepresentationData
+): string[] {
+  const supportedTypes: string[] = [];
+
+  Object.keys(representationData).forEach((representationType) => {
+    const representationTypeData = representationData[representationType];
+
+    let validateFn;
+    switch (representationType) {
+      case SegmentationRepresentations.Labelmap:
+        validateFn = validateLabelmap;
+        break;
+      // Todo: add validation for other representation types
+    }
+
+    if (validateFn) {
+      try {
+        validateFn(representationTypeData);
+        supportedTypes.push(representationType);
+      } catch (error) {
+        console.warn(
+          `Validation failed for labelmap of type ${representationType}`
+        );
+      }
+    } else {
+      supportedTypes.push(representationType);
+    }
+  });
+
+  return supportedTypes;
+}
+
+async function canConvertFromTo(fromRepresentationType, toRepresentationType) {
+  const availablePaths = conversionPaths.get(fromRepresentationType);
```

Comment:
return conversionPaths.get(fromRepresentationType)?.has(toRepresentationType) || false;
is a one line version of this if you want.